### PR TITLE
Moved nanoid use from next-server to next/build

### PR DESCRIPTION
**Pull request by slaskis**
Another (more proper?) solution to #5427

Is this what you had in mind @timneutkens ?

### CODE REVIEW

File: packages/next/build/index.js [hunk 1]
```diff
@@ -11,9 +12,14 @@ const writeFile = promisify(fs.writeFile)
 
 export default async function build (dir, conf = null) {
   const config = loadConfig(PHASE_PRODUCTION_BUILD, dir, conf)
-  const buildId = await config.generateBuildId() // defaults to a uuid
   const distDir = join(dir, config.distDir)
 
+  let buildId = await config.generateBuildId() // defaults to a uuid
+  if (buildId == null) {
```
Comment (slaskis) [inline packages/next/build/index.js]: `==` is intentional here to catch `null` or `undefined` (either would mess up any potential string methods)

or would you prefer to make this more explicit like `typeof buildId == "string"`?

**Review by timneutkens — CHANGES_REQUESTED** [review]
Could just be me, but shouldn't there be a package.json update too?

**Comment by slaskis** [comment]
I don't think so. The reason for #5427 was to move nanoid to server but this PR just leaves it where it was.

**Comment by timneutkens** [comment]
Ah I see 👍

**Comment by timneutkens** [comment]
Thanks!

**Pull request merged**